src/library/scala/collection/IndexedSeqView.scala

```diff
@@ -96,23 +96,20 @@ object IndexedSeqView {
         r
       } else Iterator.empty.next()
 
-    override def drop(n: Int): Iterator[A] = {
-      if (n > 0) {
-        pos -= n
-        remainder = Math.max(0, remainder - n)
+    override protected def sliceIterator(from: Int, until: Int): Iterator[A] = {
+      if (_hasNext) {
+        if (remainder <= from) remainder = 0
+        else if (from <= 0) {
+          if (until >= 0 && until < remainder) remainder = until
+        }
+        else {
+          pos = pos - from
+          if (until >= 0 && until < remainder) remainder = until - from
+          else remainder -= from
```

Comment:
this doesn't seem right to me. it seems to me that, supposing `until >= from` (which is ideally always), it should always be the case that `remainder = math.max(0, remainder - (until - from))`. with this code, supposing `remainder = 5`, `sliceIterator(1, 2)` will yield `remainder = 4`, when we want `remainder = 1`.

it may even be possible to replace this `if`/`else` with just `remainder = math.max(0, remainder - (until - from))`

Reply:
I did learn the previous code, BTW. I'll comment the current edit to see if it is really that simple and also easy to maintain.

Reply:
`assert(v.reverseIterator.take(5).slice(1,2).size == 1)`. `until - from`.